### ISSUE TITLE
Questions about Follower Kolbot

### ISSUE THREAD

**Issue by ARkaine2** [kolton/d2bot-with-kolbot#1641]
Is it possible to make YOU (profile which you play by your own hands) as leader (or a rusher in terminology of Kolbot) and others (other profiles) to follow you (your commands)?
How to do it?

I heard that it was fixed on official Battle.Net (by the way is that true?), but I'm interested in doing this in version 1.13(d)

Thanks.

**Comment by RNCTX** [comment]
https://github.com/kolton/d2bot-with-kolbot/wiki/Follower

You have to tell the following toon when to take TPs and get WPs, otherwise it will follow you and fight whatever you fight based on the attack config.  I use it every reset to level two toons at the same time with, works great.

**Comment by ARkaine2** [comment]
Thanks for answering.
I tried to follow this detailed guide, but i got such error:

[Error (1)] File(kolbot\libs\common\config.js:22) ReferenseError: CustomConfig is not Defined

Here is what config.js is containing in 22 (22--34 lines):

				for (n in CustomConfig) {
					if (CustomConfig.hasOwnProperty(n)) {
						if (CustomConfig[n].indexOf(me.profile) > -1) {
							if (notify) {
								print("ÿc2Loading custom config: ÿc9" + n + ".js");
							}

							configFilename = n + ".js";

							break;
						}
					}
				}

**Comment by ARkaine2** [comment]
Thanks for answering!

I followed guide but i can't get one paragraph.

Guide tells:

> Remember to set you "Attack config" properly for leecher(s) in Character config
> 
> Here is a list of full commands:
> 
> ###Main Commands###
> 
> 1 - take leader's tp from town / move to leader's town
> 2 - take leader's tp to town
> 3 - town manager
> c - get corpse
> p - pick items
> s - toggle stop
> (charname) s - toggle stop (charname)

But how should I place these commands?
I mean: What excactly should I write into Attack Config in the Character Config?
That was the first part of my question.

And the second part of my question is:
Follover doesn't follow leader even without setted commands.
Is it possible to make follower follow the leader?

**Comment by TheJMC17** [comment]
I can give my two cents about attacks in character config.
There are templates you can use: "branches\patch-113d\d2bs\kolbot\libs\config\Templates\Attacks.txt"
and you'll notice they match the text file you mentioned "branches\patch-113d\d2bs\kolbot\sdk\skills.txt"

As for the general questions you're having, seems like you're confusing bot commands with attack configs.
Bot commands (1,2,3 wp, a1,a2,a3 etc..) are typed in-chat; in-game. (from leecher to leader)

And your last question,
in your character config, in the "leeching section", at the very bottom, you'll find:
"Scripts.Follower = false; // Script that follows a manually played leader around like a merc. For a list of commands, see Follower.js"
I believe that's what you are looking for. Don't forget to set the leader's name.

Personally I'm running in single player, I wouldn't want to confuse you with my instructions.
Here's what I think should still matter for B.Net:
My leecher's character config, in "Public game options", has "Config.PublicMode = 1; // 1 = invite and accept, 2 = accept only, 3 = invite only, 0 = disable"
From the "leeching section",
It has "Scripts.Follower = false;" and "Config.Leader = "Marina";"  (Which is the name of my leader character) and I turned all Chicken settings off:

> 
	// Chicken settings
	Config.LifeChicken = 0; // Exit game if life is less or equal to designated percent.
	Config.ManaChicken = 0; // Exit game if mana is less or equal to designated percent.
	Config.MercChicken = 0; // Exit game if merc's life is less or equal to designated percent.
	Config.TownHP = 0; // Go to town if life is under designated percent.
	Config.TownMP = 0; // Go to town if mana is under designated percent.

In my leader's character config, I also have "Config.PublicMode = 1;"
From the "leeching section"
It has "Scripts.Follower = true;" and "Config.Leader = "Freya";"  (Which is the name of my leecher character)

This will, in effect, make my 97 Sorceress follow around my low level character and kill everything in its path. I control the low level manually, running around and watching my high level bot killing everything.
Not ideal but in my case, that's the best I can get without debugging some scripts.

**Comment by ARkaine2** [comment]
Thanks. I'm moving forward in understanding the follower.

> As for the general questions you're having, seems like you're confusing bot commands with attack configs.
> Bot commands (1,2,3 wp, a1,a2,a3 etc..) are typed in-chat; in-game. (from one character to another)

I typed these commands when follower wasn't in party -- they didn't affect it, but when I took party follower started to behave like puppet. :D

So I got additional question:
Is there a way to make follower automaticly take party (let's say incoming party invite from leader)?

And thanks a lot for your detailed clarification.
